src/config/config.c

```diff
@@ -1341,22 +1341,29 @@ void readFTLconf(struct config *conf, const bool rewrite)
 	initConfig(conf);
 
 	// First try to read TOML config file
-	if(readFTLtoml(NULL, conf, NULL, rewrite, NULL))
+	// If we cannot parse /etc/pihole.toml (due to missing or invalid syntax),
+	// we try to read the rotated files in /etc/pihole/config_backup starting at
+	// the most recent one and going back in time until we find a valid config
+	for(unsigned int i = 0; i < PLAIN_ROTATIONS; i++)
```

Comment:
You only include the 3 plain rotation files. But we have 15 backups (3 plain + 12 compressed). I wonder why we should omit the compressed ones? Or in general: if we have the easy chance to rescure from 15 instead of 3 I'd say we drop the compression. It does not save a lot of space to be honest:

```
-rw-r----- 1 pihole pihole  44K 19. Nov 20:58 pihole.toml.1
-rw-rw---- 1 pihole pihole  15K 18. Nov 23:49 pihole.toml.10.gz
-rw-rw---- 1 pihole pihole  15K 18. Nov 23:49 pihole.toml.11.gz
-rw-rw---- 1 pihole pihole  15K 18. Nov 21:19 pihole.toml.12.gz
-rw-rw---- 1 pihole pihole  15K 17. Nov 12:17 pihole.toml.13.gz
-rw-rw---- 1 pihole pihole  15K 16. Nov 22:11 pihole.toml.14.gz
-rw-rw---- 1 pihole pihole  40K 22. Okt 19:53 pihole.toml.15
-rw-rw---- 1 pihole pihole  15K 16. Nov 21:55 pihole.toml.15.gz
-rw-r----- 1 pihole pihole  44K 19. Nov 20:48 pihole.toml.2
-rw-r----- 1 pihole pihole  44K 19. Nov 09:20 pihole.toml.3
-rw-r--r-- 1 pihole pihole  15K 19. Nov 20:58 pihole.toml.4.gz
-rw-r--r-- 1 pihole pihole  15K 19. Nov 20:48 pihole.toml.5.gz
-rw-r--r-- 1 pihole pihole  15K 19. Nov 09:20 pihole.toml.6.gz
-rw-rw---- 1 pihole pihole  15K 18. Nov 23:57 pihole.toml.7.gz
-rw-rw---- 1 pihole pihole  15K 18. Nov 23:53 pihole.toml.8.gz
-rw-rw---- 1 pihole pihole  15K 18. Nov 23:49 pihole.toml.9.gz

```

~ 30KB per File > 360 KB

Reply:
You are right. It is not worth the extra code to inflate (decompress) them on-the-fly when we can simply wave deflation (compression) at such low costs altogether. Makes it also easier to use CLI tools like `diff` if you want to manually inspect your rotated config files.